conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala

```diff
@@ -64,16 +64,20 @@ class EthereumPersistence[F[_]: Concurrent] extends ConseilLogSupport {
     *
     * @param contractAccounts contract account eth_getBalance data
     */
-  def createContractAccounts(contractAccounts: List[Account]): DBIO[Option[Int]] =
-    Tables.Accounts ++= contractAccounts.map(_.convertTo[Tables.AccountsRow])
+  def createContractAccounts(contractAccounts: List[Account]): DBIO[Option[Int]] = {
+    import tech.cryptonomic.conseil.common.sql.CustomProfileExtension.api._
+    Tables.Accounts.insertOrUpdateAll(contractAccounts.map(_.convertTo[Tables.AccountsRow]))
+  }
 
   /**
     * Create [[DBIO]] seq with account balances.
     *
     * @param accounts account eth_getBalance data
     */
-  def createAccountBalances(accounts: List[Account]): DBIO[Option[Int]] =
-    Tables.AccountsHistory ++= accounts.map(_.convertTo[Tables.AccountsHistoryRow])
+  def createAccountBalances(accounts: List[Account]): DBIO[Option[Int]] = {
```

Comment:
I'm not sure if its correct doing insertOrUpdateAll on AccountsHistory. This table should contain all of the account states in time, and I don't think we should be doing any updates there or am I missing something?

Reply:
You're right, the problem appeared when within one block there was a contract deployed and there were some transactions on it. Now I'm filtering out contract addresses that were deployed in a block when saving accounts_history data

``
Stream
    .emits(txs)
    .filter(t => List(Some(t.from), t.to).forall(addr => !receipts.map(_.contractAddress).contains(addr)))
```